ssl/quic/quic_channel.c

```diff
@@ -2714,8 +2818,23 @@ void ossl_quic_channel_on_new_conn_id(QUIC_CHANNEL *ch,
     }
 
     if (new_remote_seq_num > ch->cur_remote_seq_num) {
+        /*
+         * Remove our existing token, failure here indicates it wasn't present
+         * which means there wasn't a stateless reset token, so we can just ignore.
+         */
+        if (!chan_update_reset_token(ch, f->stateless_reset_token,
+                                     sizeof(f->stateless_reset_token))) {
+            ossl_quic_channel_raise_protocol_error(
+                    ch, QUIC_ERR_CONNECTION_ID_LIMIT_ERROR,
+                    OSSL_QUIC_FRAME_TYPE_NEW_CONN_ID,
+                    "unable to store stateless reset token");
+
+            return;
+        }
         ch->cur_remote_seq_num = new_remote_seq_num;
         ch->cur_remote_dcid = f->conn_id;
+        memcpy(ch->cur_remote_reset_token.token, f->stateless_reset_token,
```

Comment:
Since `chan_update_reset_token` updates this isn't this superfluous? (and broken when we implement obfuscation)

Reply:
Yes, nice catch.